tests/basic/storage_pool_test.go

```diff
@@ -5168,23 +4994,201 @@ var _ = Describe("{ResizePoolDrivesInDifferentSize}", func() {
+
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			log.InfoD("Setting pools in maintenance on node %s", nodeSelected.Name)
+			err = Inst().V.EnterPoolMaintenance(nodeSelected)
+			log.FailOnError(err, "failed to set pool maintenance mode on node %s", nodeSelected.Name)
+
+			time.Sleep(1 * time.Minute)
+			expectedStatus := "In Maintenance"
+			err = waitForPoolStatusToUpdate(nodeSelected, expectedStatus)
+			log.FailOnError(err, fmt.Sprintf("node %s pools are not in status %s", nodeSelected.Name, expectedStatus))
+
+			err = Inst().V.DeletePool(nodeSelected, poolIDToDelete)
+			log.FailOnError(err, "failed to delete poolID %s on node %s", poolIDToDelete, nodeSelected.Name)
+
+			err = Inst().V.ExitPoolMaintenance(nodeSelected)
+			log.FailOnError(err, "failed to exit pool maintenance mode on node %s", nodeSelected.Name)
+
+			err = Inst().V.WaitDriverUpOnNode(nodeSelected, 5*time.Minute)
+			log.FailOnError(err, "volume driver down on node %s", nodeSelected.Name)
+
+			expectedStatus = "Online"
+			err = waitForPoolStatusToUpdate(nodeSelected, expectedStatus)
+			log.FailOnError(err, fmt.Sprintf("node %s pools are not in status %s", nodeSelected.Name, expectedStatus))
+
+			poolsAfr, err := Inst().V.ListStoragePools(metav1.LabelSelector{})
+			log.FailOnError(err, "Failed to list storage pools")
+
+			dash.VerifySafely(len(poolsBfr) > len(poolsAfr), true, "verify pools count is updated after pools deletion")
+
+			poolsMap, err = Inst().V.GetPoolDrives(&nodeSelected)
+			log.FailOnError(err, "error getting pool drive from the node [%s]", nodeSelected.Name)
+			_, ok := poolsMap[poolIDToDelete]
+			dash.VerifyFatal(ok, false, "verify drive is deleted from the node")
+
+		})
+
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("resiznoqr-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
 
-		stepLog = "destroy all the applications created before test runs"
+		newSpecSize := (poolToDelete.TotalSize / units.GiB) / 2
+		///creating a spec to perform add  drive
+		driveSpecs, err := GetCloudDriveDeviceSpecs()
+		log.FailOnError(err, "Error getting cloud drive specs")
+
+		deviceSpec := driveSpecs[0]
+		deviceSpecParams := strings.Split(deviceSpec, ",")
+
+		paramsArr := make([]string, 0)
+		for _, param := range deviceSpecParams {
+			if strings.Contains(param, "size") {
+				paramsArr = append(paramsArr, fmt.Sprintf("size=%d,", newSpecSize))
+			} else {
+				paramsArr = append(paramsArr, param)
+			}
+		}
+		newSpec := strings.Join(paramsArr, ",")
+		stepLog = fmt.Sprintf("Adding cloud drive to node %s with size %s", nodeSelected.Name, newSpec)
+
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			err = Inst().V.AddCloudDrive(&nodeSelected, newSpec, -1)
+			log.FailOnError(err, "error adding new drive to node %s", nodeSelected.Name)
+			log.InfoD("Validate pool rebalance after drive add")
```

Comment:
nit: Can we also include node name here `nodeSelected.Name` like you did for the error

Reply:
fixed